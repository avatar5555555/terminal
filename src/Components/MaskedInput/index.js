// @flow

import React from 'react'
import Input from '@material-ui/core/Input'
import InputLabel from '@material-ui/core/InputLabel'
import FormHelperText from '@material-ui/core/FormHelperText'
import FormControl from '@material-ui/core/FormControl'
import InputMask from 'react-input-mask'
import { FormattedMessage } from 'react-intl'

type Props = {
  mask: string,
  id: string,
  value: string | number,
  onChange: Function,
  onBlur: Function,
  error: string,
  valid: boolean,
  className: string
}

const MInput = props => (
  <InputMask {...props} alwaysShowMask={false} maskChar="" />
)

const MaskedInput = (props: Props) => {
  const { mask, id, valid, error, className, ...rest } = props
  return (
    <FormControl className={className}>
      <InputLabel htmlFor={id} error={!valid}>
        <FormattedMessage id={`Form.${id}`} />
      </InputLabel>
      <Input
        {...rest}
        id={id}
        inputComponent={MInput}
        inputProps={{ mask }}
        error={!valid}
      />
      {!valid && (
        <FormHelperText error={!valid} id={`${id}-helper-text`}>
          <FormattedMessage id={`Errors.${error}`} />
        </FormHelperText>
      )}
    </FormControl>
  )
}

export default MaskedInput
