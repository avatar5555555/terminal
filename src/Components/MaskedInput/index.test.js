import React from 'react'
import { shallow } from 'enzyme'

import MaskedInput, { MInput } from '.'

const props = {
  mask: 'string',
  id: 'string',
  value: '',
  onChange: jest.fn(),
  onBlur: jest.fn(),
  error: '',
  valid: true,
  className: 'string'
}

describe('MInput', () => {
  it('renders', () => {
    const wrapper = shallow(<MInput />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('MaskedInput', () => {
  const wrap = (props = {}) => shallow(<MaskedInput {...props} />)

  it('renders', () => {
    expect(wrap(props)).toMatchSnapshot()
  })

  it('renders with valid === false', () => {
    expect(wrap({ ...props, valid: false })).toMatchSnapshot()
  })
})
