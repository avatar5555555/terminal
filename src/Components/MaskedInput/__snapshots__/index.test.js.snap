// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MInput renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alwaysShowMask": false,
      "maskChar": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alwaysShowMask": false,
        "maskChar": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MaskedInput renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MaskedInput
    className="string"
    error=""
    id="string"
    mask="string"
    onBlur={[Function]}
    onChange={[Function]}
    valid={true}
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(InputLabel)
          error={false}
          htmlFor="string"
>
          <FormattedMessage
                    id="Form.string"
                    values={Object {}}
          />
</WithStyles(InputLabel)>,
        <WithStyles(Input)
          error={false}
          id="string"
          inputComponent={[Function]}
          inputProps={
                    Object {
                              "mask": "string",
                            }
          }
          onBlur={[Function]}
          onChange={[Function]}
          value=""
/>,
        false,
      ],
      "className": "string",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FormattedMessage
            id="Form.string"
            values={Object {}}
/>,
          "error": false,
          "htmlFor": "string",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "Form.string",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": false,
          "id": "string",
          "inputComponent": [Function],
          "inputProps": Object {
            "mask": "string",
          },
          "onBlur": [Function],
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(InputLabel)
            error={false}
            htmlFor="string"
>
            <FormattedMessage
                        id="Form.string"
                        values={Object {}}
            />
</WithStyles(InputLabel)>,
          <WithStyles(Input)
            error={false}
            id="string"
            inputComponent={[Function]}
            inputProps={
                        Object {
                                    "mask": "string",
                                  }
            }
            onBlur={[Function]}
            onChange={[Function]}
            value=""
/>,
          false,
        ],
        "className": "string",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FormattedMessage
              id="Form.string"
              values={Object {}}
/>,
            "error": false,
            "htmlFor": "string",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "Form.string",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": false,
            "id": "string",
            "inputComponent": [Function],
            "inputProps": Object {
              "mask": "string",
            },
            "onBlur": [Function],
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MaskedInput renders with valid === false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MaskedInput
    className="string"
    error=""
    id="string"
    mask="string"
    onBlur={[Function]}
    onChange={[Function]}
    valid={false}
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(InputLabel)
          error={true}
          htmlFor="string"
>
          <FormattedMessage
                    id="Form.string"
                    values={Object {}}
          />
</WithStyles(InputLabel)>,
        <WithStyles(Input)
          error={true}
          id="string"
          inputComponent={[Function]}
          inputProps={
                    Object {
                              "mask": "string",
                            }
          }
          onBlur={[Function]}
          onChange={[Function]}
          value=""
/>,
        <WithStyles(FormHelperText)
          error={true}
          id="string-helper-text"
>
          <FormattedMessage
                    id="Errors."
                    values={Object {}}
          />
</WithStyles(FormHelperText)>,
      ],
      "className": "string",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FormattedMessage
            id="Form.string"
            values={Object {}}
/>,
          "error": true,
          "htmlFor": "string",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "Form.string",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": true,
          "id": "string",
          "inputComponent": [Function],
          "inputProps": Object {
            "mask": "string",
          },
          "onBlur": [Function],
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FormattedMessage
            id="Errors."
            values={Object {}}
/>,
          "error": true,
          "id": "string-helper-text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "Errors.",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(InputLabel)
            error={true}
            htmlFor="string"
>
            <FormattedMessage
                        id="Form.string"
                        values={Object {}}
            />
</WithStyles(InputLabel)>,
          <WithStyles(Input)
            error={true}
            id="string"
            inputComponent={[Function]}
            inputProps={
                        Object {
                                    "mask": "string",
                                  }
            }
            onBlur={[Function]}
            onChange={[Function]}
            value=""
/>,
          <WithStyles(FormHelperText)
            error={true}
            id="string-helper-text"
>
            <FormattedMessage
                        id="Errors."
                        values={Object {}}
            />
</WithStyles(FormHelperText)>,
        ],
        "className": "string",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FormattedMessage
              id="Form.string"
              values={Object {}}
/>,
            "error": true,
            "htmlFor": "string",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "Form.string",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": true,
            "id": "string",
            "inputComponent": [Function],
            "inputProps": Object {
              "mask": "string",
            },
            "onBlur": [Function],
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FormattedMessage
              id="Errors."
              values={Object {}}
/>,
            "error": true,
            "id": "string-helper-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "Errors.",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
